includes/payment-methods/class-affirm-payment-method.php

```diff
@@ -39,6 +39,7 @@ public function __construct( $token_service ) {
 				'max' => 3000000,
 			], // Represents USD 50 - 30,000 USD.
 		];
+		$this->countries           = [ 'US', 'CA' ];
```

Comment:
Two things here.

From the code perspective. I followed same approach we have for currencies. Empty array means that there are no restrictions. Otherwise we'd have to list all countries that we support and update that list every time we add new supported country to WCpay. 

You also had comment 
> Can we do a bit of optimization and ensure supportedCountries are not empty in the first place?

which is related to this. We need a way to figure out, if any restriction logic should be applied. 

I think the approach is good, but we should ensure that it's clear why we have to check for supportedCountries. Either wrap in proper method name or adding a comment would work.

Reply:
Here's a simple example of what I meant about "Either wrap in proper method name or adding a comment would work."
```JS
const isRestrictedInAnyCountry = !! upeConfig.countries.length;
```

Now it's clear that if there are no countries specified, then we allow PM in all countries.

Reply:
Makes sense, `const isRestrictedInAnyCountry = !! upeConfig.countries.length;` will work as bring the needed clarity. Go for it =)

Reply:
Before merging, lets polish as per remaining comments.